### ISSUE TITLE
2nd page base64 error in Reports

### ISSUE THREAD

**Issue by mbates14** [salesagility/SuiteCRM#3067]
My case was closed! Blamed it on the PHP.INI file which its NOT my PHP.INI file. its CLEARLY A BUG!!! in the CRM core system somewhere. I cannot get to a 2nd page. Even suppressing the error in the php.ini file does nothing as instead of showing a 2nd page of results, it just drops out with 0 of 0 again...

ITS A BUG! Dont friggen close my request and not allow me to defend myself? Sure blame it on the user and give up. typical. Also, none of my posts are showing up in the forum, so I cant go there! Says no topics, and posts, and I know for a FACT I made posts. So I assume they have been stuck in "moderation" for days on end. 

![b83739e2-ed23-11e6-8477-162f0955e774](https://cloud.githubusercontent.com/assets/22461464/22763143/f43e8ef0-ee30-11e6-9cc4-892686d1b373.png)

**Comment by shogunpol** [comment]
@mbates14 , this is not seen as an issues in testing environment, we always take care seriously of all issues. Can you provide extended steps to reproduce it, including the report conditions and fields.

**Comment by horus68** [comment]
@mbates14 just an user here, but I can assure you that when an issue is closed by admins you can always request to be reopened posting a new comment and reference the user that closed it with an @ ...

**Comment by mbates14** [comment]
Sorry, This is the first time I have used this github system. I am absolutely unfamiliar with it, it seems more complicated than it should be IMHO. 

As far as reproducing it, All I do is click on the Sales Report in our system, and it lists accounts. Click 2nd page, boom crash. Thats it. I can screenshot the edit screen of the report if that helps. Also the graph is messed up as well, all the text is run together for some reason.

**Comment by mbates14** [comment]
Here is the graph issue 
![graph error](https://cloud.githubusercontent.com/assets/22461464/22835224/046e89dc-ef86-11e6-887b-7d3da9c6da2d.png)

Here is the Report information: 
![report](https://cloud.githubusercontent.com/assets/22461464/22835260/0ebe6916-ef86-11e6-85bc-befad8bd0c11.png)

**Comment by mbates14** [comment]
One more thing. 

Soon as I click sales report, I get the graph as shown above. But I also get everything on "Page 1" 
As shown here:
![page 1](https://cloud.githubusercontent.com/assets/22461464/22835744/d0109ba6-ef87-11e6-9c7d-12785feb43f7.png)

Which results in what I posted above. Here:
![page 2](https://cloud.githubusercontent.com/assets/22461464/22835755/d7a176ce-ef87-11e6-8391-1072a459be0f.png)

**Comment by mbates14** [comment]
If this helps, This is in the log:
Fri Feb 10 14:24:17 2017 [32590][1][FATAL]  Query Failed: SELECT count(*) c FROM (SELECT `opportunities`.id AS 'opportunities_id', `opportunities`.name AS 'Opportunity_Name0', DATE_FORMAT(`opportunities`.date_entered - INTERVAL 300 MINUTE,'%Y-%m') AS 'Date_Created1', `opportunities`.assigned_user_id AS 'Assigned_to2', `accounts`.id AS 'accounts_id', `accounts`.id AS 'Account_Name3', `opportunities`.currency_id AS 'opportunities_currency_id', `opportunities`.amount AS 'Opportunity_Amount4', DATE_FORMAT(`opportunities`.date_closed,'%Y-%m') AS 'Expected_Close_Date5', `opportunities`.next_step AS 'Next_Step6', `opportunities`.probability AS 'Probability_(%)7' FROM `opportunities` LEFT JOIN accounts_opportunities `opportunities|accounts` ON `opportunities`.id=`opportunities|accounts`.opportunity_id AND `opportunities|accounts`.deleted=0

LEFT JOIN accounts `accounts` ON `accounts`.id=`opportunities|accounts`.account_id AND `accounts`.deleted=0
 WHERE ( `opportunities`.date_closed >= DATE_ADD(opportunities.undefined, INTERVAL   ) OR `opportunities`.assigned_user_id = '' OR `opportunities`.assigned_user_id IS NULL ) AND  opportunities.deleted = 0 ) as n: MySQL error 1064: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near ') OR `opportunities`.assigned_user_id = '' OR `opportunities`.assigned_user_id I' at line 4
Fri Feb 10 14:24:17 2017 [32590][1][FATAL]  Query Failed: SELECT `opportunities`.id AS 'opportunities_id', `opportunities`.name AS 'Opportunity_Name0', DATE_FORMAT(`opportunities`.date_entered - INTERVAL 300 MINUTE,'%Y-%m') AS 'Date_Created1', `opportunities`.assigned_user_id AS 'Assigned_to2', `accounts`.id AS 'accounts_id', `accounts`.id AS 'Account_Name3', `opportunities`.currency_id AS 'opportunities_currency_id', `opportunities`.amount AS 'Opportunity_Amount4', DATE_FORMAT(`opportunities`.date_closed,'%Y-%m') AS 'Expected_Close_Date5', `opportunities`.next_step AS 'Next_Step6', `opportunities`.probability AS 'Probability_(%)7' FROM `opportunities` LEFT JOIN accounts_opportunities `opportunities|accounts` ON `opportunities`.id=`opportunities|accounts`.opportunity_id AND `opportunities|accounts`.deleted=0

LEFT JOIN accounts `accounts` ON `accounts`.id=`opportunities|accounts`.account_id AND `accounts`.deleted=0
 WHERE ( `opportunities`.date_closed >= DATE_ADD(opportunities.undefined, INTERVAL   ) OR `opportunities`.assigned_user_id = '' OR `opportunities`.assigned_user_id IS NULL ) AND  opportunities.deleted = 0  LIMIT 20,20: MySQL error 1064: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near ') OR `opportunities`.assigned_user_id = '' OR `opportunities`.assigned_user_id I' at line 4

**Comment by mbates14** [comment]
Seems the problem is a missing parameter after INTERVAL, I copied and pasted the entire thing into SQL inside phpMyAdmin, and Replacing the "missing" argument after INTERVAL in Date_Add, to something arbtrary like 30 DAYS, it worked. That is definitely a code issue! 

I am running PHP 5.4.16 and Mysql 5.7.17 something. I went ahead and updated PHP to 5.5.38, and SQL is still the same. 

Doing that made no difference, and I also went ahead and set the SQL_MODE in the my.cnf to blank to get rid of zero, and strict mode, etc. again no differences. 

Last ditch effort is to blow out the server completely, and re-install from scratch. My fear is soon as I restore the database, its gonna bring the problem right back.

**Comment by mbates14** [comment]
I downgraded mysql, and upgraded PHP, at the same time I backed up only my tables from the DB which had core data in it. Such as accounts, cases, etc... No feeds, or audits, cstm, or anything of that nature that gets built and shuffled during runtime. 

Git cloned this repository and reinstalled the whole system. I brought all my tables back over through phpMyAdmin, and that fixed all my existing issues EXCEPT the sales report error! But when I deleted and re-added the conditional logic, it started working. (I have done this before a multitude of times on the previous install). 

So something was broken somewhere that got fixed by blowing out the DB, and CRM folders and doing a full re-install from this repository and I brought my data back over on SQL import. Go figure. the Graph is still screwed up though, but everything else works. Except links show in print.
